os/booting-on-vmware.md

```diff
@@ -76,7 +76,38 @@ Cloud-config data can be passed into a VM by attaching a [config-drive][config-d
 
 The config-drive standard was originally an OpenStack feature, which is why you'll see the string `openstack` in a few bits of configuration. This naming convention is retained, although CoreOS supports config-drive on all platforms.
 
-The cloud-config `$private_ipv4` and `$public_ipv4` substitution variables are supported on VMWare in CoreOS releases 801.0.0 and greater.
+This example will configure CoreOS components: etcd2, fleetd and flannel. `$private_ipv4` and `$public_ipv4` are supported on VMWare in CoreOS releases 801.0.0 and greater, and **only** when you explicitly configure interfaces' roles to `private` or `public` in [Guestinfo][guestinfo] for each VMWare instance.
+
+```yaml
+#cloud-config
+
+# include one or more SSH public keys
+ssh_authorized_keys:
+  - ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDGdByTgSVHq.......
+coreos:
+  etcd2:
+    # generate a new token for each unique cluster from https://discovery.etcd.io/new?size=3
+    # specify the initial size of your cluster with ?size=X
+    discovery: https://discovery.etcd.io/<token>
+    advertise-client-urls: http://$private_ipv4:2379,http://$private_ipv4:4001
+    initial-advertise-peer-urls: http://$private_ipv4:2380
+    # listen on both the official ports and the legacy ports
+    # legacy ports can be omitted if your application doesn't depend on them
+    listen-client-urls: http://0.0.0.0:2379,http://0.0.0.0:4001
+    listen-peer-urls: http://$private_ipv4:2380
+  units:
+    - name: etcd2.service
+      command: start
+    - name: fleet.service
+      command: start
+    - name: flanneld.service
```

Comment:
for consistency with the rest of our cloudconfig examples, i'd say don't set up flannel